### PR TITLE
Add workaround for NavigationView ItemSource and template

### PR DESCRIPTION
**Pull request by jeromelaban**
GitHub Issue (If applicable): https://github.com/unoplatform/uno/issues/2477

## PR Type

What kind of change does this PR introduce?
- Bugfix

## What is the new behavior?

Setting an `NavigationView.ItemsSource` now partially works, but the selection cannot be visualized until https://github.com/unoplatform/uno/issues/2477 is fixed.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current [supported SDKs](https://github.com/unoplatform/uno/blob/master/README.md#uno-features)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] [Wasm UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Updated the [Release Notes](https://github.com/unoplatform/uno/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/NavigationViewTests/NavigationView_Pane_Automated.xaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+﻿<Page x:Class="SamplesApp.Samples.Windows_UI_Xaml_Controls.NavigationViewTests.NavigationView_Pane_Automated"
```
Comment (agneszitte-nventive) [inline src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/NavigationViewTests/NavigationView_Pane_Automated.xaml]: Ident ;)

File: src/Uno.UI/UI/Xaml/Controls/NavigationView/NavigationViewList.Uno.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+		}
+
+		internal override int IndexFromContainerInner(DependencyObject container)
+		{
+			var index = base.IndexFromContainerInner(container);
+
+			if(index == -1)
+			{
+				if(
+					container is FrameworkElement fe
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/Controls/NavigationView/NavigationViewList.Uno.cs]: These `if`s can be combined

File: src/Uno.UI/Uno.UI.csproj [hunk 3]
```diff
@@ -362,6 +357,10 @@
 		</ProjectReference>
 	</ItemGroup>
 
+	<ItemGroup>
+	  <UpToDateCheckInput Remove="UI\Xaml\Controls\NavigationView\NavigationViewList.Uno.cs" />
```
Comment (davidjohnoliver) [inline src/Uno.UI/Uno.UI.csproj]: Not needed

File: src/Uno.UI/UI/Xaml/Controls/NavigationView/NavigationViewList.Uno.cs [hunk 4]
```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+//
+// This file is a C# translation of the NavigationViewList.cpp file from WinUI controls.
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/Controls/NavigationView/NavigationViewList.Uno.cs]: Remove this header ;)

**Pull request merged**